extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/build/CacheInfo.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.cache.runtime.build;
+
+import java.time.Duration;
+import java.util.Objects;
+
+public class CacheInfo {
```

Comment:
Would this means, long term, all providers would be expected to offer `CacheInfo`-level of feature support? As in, is this the baseline configuration like JCache's `CacheConfiguration`?

JCache tried to dictate the baseline configuration, which created a mess by being limited yet claiming you could configure the cache without implementation-specific code (e.g. no maximumSize). Spring Cache doesn't make any assumptions, other than basic dispatching, which is much better. In their case they use `CaffeineSpec` for configuration that can be set in a conf file. They also allow binding in Java code custom builder instances, e.g. so that `Caffeine.ticker` could be set for unit testing expiration via Guava's `FakeTicker`.

Reply:
@ben-manes

You made a very good point here and it took me quite some time before I could improve this subject, but I just pushed a refactored version of the cache configuration. I tried to stick to the Quarkus configuration philosophy which is to always favor the use of `applications.properties`. That's why I introduced providers-specific configuration keys in `CacheConfig` and also made the previous `CacheInfo` class a provider-specific class.

Here's an example of a possible `applications.properties` content with these changes:
```
quarkus.cache.type=caffeine
quarkus.cache.caffeine.namespace."fooCache".initial-capacity=10
quarkus.cache.caffeine.namespace."fooCache".maximum-size=100
quarkus.cache.caffeine.namespace."barCache".maximum-size=1000
```

Does it look better that way?

Reply:
Thanks! That makes sense.

There is a `CaffeineSpec` if helpful, but mirror's Guava's `CacheBuilderSpec` so some styling is different (e.g. camel case). But at least shows which are externalizable settings. In my work code I use Typesafe Config which is also hyphen case, so I have a tiny parser to translate the format into the spec. That might be easier for you if you want to support more options.